domain/application/service/unit_test.go

```diff
@@ -207,3 +212,102 @@ func (s *unitServiceSuite) TestUpdateCAASUnitNotAlive(c *gc.C) {
 	err := s.service.UpdateCAASUnit(context.Background(), coreunit.Name("foo/666"), UpdateCAASUnitParams{})
 	c.Assert(err, jc.ErrorIs, applicationerrors.ApplicationNotAlive)
 }
+
+// TestSetReportedUnitAgentVersionInvalid is here to assert that if pass a
+// junk agent binary version to [Service.SetReportedUnitAgentVersion] we get
+// back an error that satisfies [coreerrors.NotValid].
```

Comment:
Hi I think this should be coreerrors.NotSupported?

Reply:
This is to pass a junk agent binary version, which should be consistent with not valid?